packages/core/lib/types.ts

```diff
@@ -72,7 +78,16 @@ export interface ProcessRequestOptions<TContext, TRootValue> {
   /**
    * An optional function which will be used to execute instead of default `execute` from `graphql-js`.
    */
-  execute?: (...args: any[]) => any;
+  execute(
```

Comment:
`typeof execute`?

Reply:
`execute` is an overloaded function in `graphql-js`  if we do it exactly like that that would make it the same signature but then as use someone extending it you will need to do similar thing to do checks if `args` is length 1 or greater that would be writing more boilerplate to just extend it. (Same for `parse`) 

```ts
export function execute(args: ExecutionArgs): PromiseOrValue<ExecutionResult>;
export function execute(
  schema: GraphQLSchema,
  document: DocumentNode,
  rootValue?: any,
  contextValue?: any,
  variableValues?: Maybe<{ [key: string]: any }>,
  operationName?: Maybe<string>,
  fieldResolver?: Maybe<GraphQLFieldResolver<any, any>>,
  typeResolver?: Maybe<GraphQLTypeResolver<any, any>>,
): PromiseOrValue<ExecutionResult>;
```

Reply:
I think we have one in `envelop` (added by @n1ru4l), see: https://github.com/dotansimha/envelop/blob/main/packages/types/src/graphql.ts#L15-L43

Reply:
Similar to envelop wouldn't it also make sense to explore `getExecuteArgs `https://github.com/dotansimha/envelop/blob/7704fc34f2b1cfe4f27b379e53a386db35516b9d/packages/core/src/utils.ts#L90-L103

Reply:
Can't we theoretically only use the object for execute syntax? It si supported in graphql-js 15 and the only way for graphql-js 16 🤔

Reply:
I think just using objects for args is better